src/snakemake/report/__init__.py

```diff
@@ -462,14 +462,17 @@ def shorten_ids(results: Mapping[Category, Mapping[Category, List[FileRecord]]])
         for subcat, catresults in subcats.items()
         for res in catresults
     ]
-    for id_len in range(8, 65):
-        if len({rec.id[:id_len] for rec in file_records}) == len(file_records):
-            for rec in file_records:
-                rec.id = rec.id[:id_len]
-            return
-    logger.warning(
-        "Obtained result IDs are non-unique. Certain results will be not accessible."
-    )
+    full_ids = [rec.id for rec in file_records]
+    shortened_ids = [rec.id[:16] for rec in file_records]
```

Comment:
Please define the 16 as a global constant here, and refer to that constant also below in the ReportHref class.